extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ObserverValidationProcessor.java

```diff
@@ -0,0 +1,63 @@
+package io.quarkus.arc.deployment;
+
+import java.util.Collection;
+
+import javax.inject.Inject;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.IndexView;
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.processor.BeanDeploymentValidator;
+import io.quarkus.arc.processor.DotNames;
+import io.quarkus.arc.processor.ObserverInfo;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ApplicationArchivesBuildItem;
+
+/**
+ * Validates observer methods from application classes.
+ * If an observer listening for {@code @Initialized(ApplicationScoped.class)} is found, it logs a warning.
+ */
+public class ObserverValidationProcessor {
+
+    private static final Logger LOGGER = Logger.getLogger(ObserverValidationProcessor.class.getName());
+
+    @Inject
+    BuildProducer<BeanDeploymentValidatorBuildItem> validators;
```

Comment:
There is no need to use an injection field for a producer. I'd just add the `BuildProducer<BeanDeploymentValidatorBuildItem> validators` next to the `applicationArchivesBuildItem` parameter.

Reply:
Yeah I know, will do that if it'd preferable.